VideoLocker/src/main/java/org/edx/mobile/task/SetThreadReadTask.java

```diff
@@ -11,22 +11,15 @@
 
 public class SetThreadReadTask extends Task<DiscussionThread> {
     private final DiscussionThread thread;
-    private final boolean read;
 
     public SetThreadReadTask(@NonNull Context context,
-                             @NonNull DiscussionThread thread, boolean read) {
+                             @NonNull DiscussionThread thread) {
         super(context);
         this.thread = thread;
-        this.read = read;
     }
 
     @Override
     public DiscussionThread call() throws RetroHttpException {
-        return environment.getDiscussionAPI().setThreadRead(thread, read);
-    }
-
-    @Override
-    protected void onSuccess(DiscussionThread discussionThread) {
-        EventBus.getDefault().post(new DiscussionThreadUpdatedEvent(discussionThread));
+        return environment.getDiscussionAPI().setThreadRead(thread, true);
```

Comment:
> Its semantically wrong to name something what it is not.

But it _is_ getting the thread too, and in fact that's the main purpose of it as far as the client code is concerned (marking it as read being a useful side-effect of the implementation).

> It might come in handy in the future since we won't be wanting to set a thread's read state, if only what we want is to GET the thread details.

I don't currently see any scenario where we might need to do that, but OK it's up to you.